src/pipelines/mod.rs

```diff
@@ -214,11 +224,31 @@ impl AssetFile {
     ///
     /// The base file name (stripped path, without any parent folders) is returned if the operation
     /// was successful.
-    pub async fn copy(&self, to_dir: &Path, with_hash: bool) -> Result<String> {
-        let bytes = fs::read(&self.path)
+    pub async fn copy(&self,
+                      to_dir: &Path,
+                      with_hash: bool,
+                      minify: bool,
+                      file_type: AssetFileType
+    ) -> Result<String> {
+        let mut bytes = fs::read(&self.path)
             .await
             .with_context(|| format!("error reading file for copying {:?}", &self.path))?;
 
+        bytes = if minify {
+            match file_type {
+                AssetFileType::Css => {
+                    minify_html::minify(&*bytes, &Cfg::spec_compliant())
+                }
+                AssetFileType::Icon => {
+                    oxipng::optimize_from_memory(bytes.as_ref(), &Options::from_preset(PNG_OPTIMIZATION_LEVEL))
+                        .expect(format!("error optimizing PNG {:?}", &self.path).as_str())
```

Comment:
So, instead of using expect, which will panic, we should avoid panics at all costs. Let's use standard error propagation with `?`. And, as we do in other places in this code base, let's use [anyhow::Context](https://docs.rs/anyhow/latest/anyhow/trait.Context.html).
```suggestion
                        .with_context(|| format!("error optimizing PNG {}", &self.path))?
```

Roughly.